### ISSUE TITLE
Trying to restore page for non-private VMA

### ISSUE THREAD

**Issue by xy815661276** [checkpoint-restore/criu#1226]
I successfully ported CRIU to the Android operating system to run before, and can successfully checkpoint & restore a loop process, and I want to migrate the process under X86_64 to the Android of the aarch64 architecture to run, here I use [criu-het](https://github.com/systems-nuts/criu-het), when I `criu-het restore -j` the following error was prompted when restoring in the Android operating system.
```shell
1|raphael:/tmp/hello/aarch64 # criu-het restore -j
Error (criu/tun.c:85): tun: Unable to create tun: No such file or directory
Warn  (criu/libnetlink.c:55): ERROR -95 reported by netlink
Error (criu/net.c:3155): Unable to create a veth pair: -95
Warn  (criu/net.c:3177): NSID isn't reported for network links
Warn  (criu/kerndat.c:881): Can't keep kdat cache on non-tempfs
 22986: Error (criu/mem.c:990): Trying to restore page for non-private VMA
 22986: Error (criu/mem.c:1120): Page entry address 7ffcd6228000 outside of VMA 7ffcd5e29000-7ffcd6629000 status 201 (task size: 8000000000)
Error (criu/cr-restore.c:1421): 22986 exited, status=1
Error (criu/cr-restore.c:2300): Restoring FAILED.
1|raphael:/tmp/hello/aarch64 # ls
cgroup.img     files.img     inventory.img     pages-1.img tty-info.img
core-22986.img fs-22986.img  mm-22986.img      pstree.img
fdinfo-2.img   ids-22986.img pagemap-22986.img seccomp.img
raphael:/tmp/hello/aarch64 #
```
Is this a VMA error?

**Comment by adrianreber** [comment]
Not sure anybody here can help you. If you have a problem with criu-het, you should probably talk to the criu-het authors.

**Comment by rst0git** [comment]
Hi @xy815661276, it looks like [HetCRIU](https://www.research.ed.ac.uk/portal/files/141999220/Edge_Computing_BARBALACE_DOA25012020_AFV.pdf) is modifying the image files between architectures. You might want to check if the [pages and pagemap images](https://criu.org/Memory_dumps) have been converted correctly.

**Comment by xy815661276** [comment]
@rst0git  Thank you very much, I will try.

**Comment by rst0git** [comment]
IIRC @rppt (who presented similar work at [LPC 2018](https://www.linuxplumbersconf.org/event/2/contributions/205/)) may be able to help with this.

**Comment by xy815661276** [comment]
The reason is that Android uses 39-bit addressing, while ubuntu uses 48-bit addressing. In order to solve this problem, I am trying to convert a 48-bit address to a 39-bit address.

**Comment by xy815661276** [comment]
The problem has been solved by using the same 48-bit addressing.
Extremely grateful.
